File: IntelliJ IDEA/Java/src/com/eruru/mvvm/swing/MVVMLabel.java
package com.eruru.mvvm.swing;

import com.eruru.mvvm.MVVMAPI;
import com.eruru.mvvm.MVVMBinding;

import javax.swing.*;

public class MVVMLabel extends MVVMControl {

	private final JLabel control;

	private MVVMBinding text;

	public MVVMLabel (JLabel control) {
		super (control);
		this.control = control;
	}

	public JLabel getControl () {
		return control;
	}

	public MVVMBinding getText () {
		return getBinding (text, this::setText);
	}

	public void setText (MVVMBinding binding) {
		text = setBinding (binding, control::getText, targetValue -> control.setText (MVVMAPI.to (targetValue, String.class)));
	}

}

File: Visual Studio/C#/Eruru.MVVM/MVVMLabel.cs
using System.Windows.Forms;

namespace Eruru.MVVM {

	public class MVVMLabel : MVVMControl {

		public new Label Control { get; }
		public MVVMBinding Text {

			get {
				return GetBinding (ref _Text, binding => Text = binding);
			}

			set {
				SetBinding (ref _Text, value, () => Control.Text, targetValue => Control.Text = MVVMAPI.To<string> (targetValue));
			}

		}

		MVVMBinding _Text;

		public MVVMLabel (Label control) : base (control) {
			Control = control;
		}

	}

}